### PR TITLE
fix(vue): render-no-results called when modifiedoptions length is zero

### PR DESCRIPTION
**Pull request by anik-ghosh-au7**
**Before submitting a pull request,** please make sure the following is done:

- [x] Describe the proposed changes and how it'll improve the library experience.

renderNoResults to also be called when modifiedOptions length is zero in the list components in vue-reactivesearch, currently, it was returning null.
 - SingleList
 - SingleDropdownList
 - MultiList
 - MultiDropdownList
 
 Look into the following issue for more reference - https://github.com/appbaseio/reactivesearch/issues/1610

- [x] Please make sure that there are no linting errors in the code.
- [x] Add a demo video/gif/screenshot to explain how did you test the fix.

https://www.loom.com/share/c071a45f34b945d290e45044eb00f088

- [ ] If it is a global change, try to add any side effects that it could have.
- [ ] Create a PR to add/update the docs (if needed) at [here](https://github.com/appbaseio/Docs).
- [ ] Create a PR to add/update the storybook (if needed) at [here](https://github.com/appbaseio/playground).

**Learn more about contributing:** [Contributing guides](https://github.com/appbaseio/reactivesearch/blob/next/.github/CONTRIBUTING.md)

### CODE REVIEW

File: packages/vue/src/components/list/MultiDropdownList.jsx [hunk 1]
```diff
@@ -175,14 +175,21 @@ const MultiDropdownList = {
...
+		const renderNoResults = this.$scopedSlots.renderNoResults || this.$props.renderNoResults;
 		let selectAll = [];
 
 		if (renderErrorCalc && this.error) {
 			return isFunction(renderErrorCalc) ? renderErrorCalc(this.error) : renderErrorCalc;
 		}
 
 		if (!this.hasCustomRenderer && this.$data.modifiedOptions.length === 0) {
-			return null;
+			if(renderNoResults && isFunction(renderNoResults)) {
```
Comment (bietkul) [inline packages/vue/src/components/list/MultiDropdownList.jsx:186]: @anik-ghosh-au7 Optimize the condition. Why are you using `div` here?

File: packages/vue/src/components/list/SingleDropdownList.jsx [hunk 2]
```diff
@@ -167,13 +167,20 @@ const SingleDropdownList = {
...
 		const renderLabelCalc = this.$scopedSlots.renderLabel || renderLabel;
+		const renderNoResults = this.$scopedSlots.renderNoResults || this.$props.renderNoResults;
 
 		if (renderErrorCalc && this.error) {
 			return isFunction(renderErrorCalc) ? renderErrorCalc(this.error) : renderErrorCalc;
 		}
 
 		if (!this.hasCustomRenderer && this.$data.modifiedOptions.length === 0) {
-			return null;
+			if(renderNoResults && isFunction(renderNoResults)) {
```
Comment (bietkul) [inline packages/vue/src/components/list/SingleDropdownList.jsx:177]: @anik-ghosh-au7 Optimize the condition. Why are you using `div` here?

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 I think you've ignored the `loading` prop. We should not render `renderNoResults` when `loading` is `true`.
Share a demo on **slow network connection** where `renderNoResults` is defined (don't change `dataField`). It shouldn't render `renderNoResults` until the request is resolved.

File: packages/vue/src/components/list/SingleDropdownList.jsx (hunk 2 above)
Comment (anik-ghosh-au7) [reply packages/vue/src/components/list/SingleDropdownList.jsx:177]: @bietkul the `<div>` is needed without it the content is not getting rendered, in the video I have shown the issue

**Comment by anik-ghosh-au7** [comment]
@bietkul I have done the updates - https://www.loom.com/share/c200110df74a4f2888d1ccb8bb4d9629

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 Fix the formatting in all the files.

**Pull request merged**